enhancements/sig-architecture/43-governance-addon-controller/README.md

```diff
@@ -0,0 +1,118 @@
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in
+      [website](https://github.com/open-cluster-management/website/)
+
+## Summary
+
+The goal of this feature is to automate and simplify deployment of the governance policy components
+to managed clusters by creating a controller that leverages the
+[addon framework](https://github.com/open-cluster-management-io/addon-framework/) in order to deploy
+components using the ManagedClusterAddOn kind.
+
+## Motivation
+
+The governance policy framework is currently deployed using individual manifest files deployed
+directly by the user. While this is functional, it does not lend itself to automation,
+customization, scalability, nor hub-centered management that Open Cluster Management demands.
+
+### Goals
+
+- Develop a controller that deploys governance policy components to managed clusters using the
+  ManagedClusterAddOn kind.
+
+### Non-Goals
+
+- None at this time.
+
+## Proposal
+
+### User Stories
+
+1. As a k8s admin, I want to deploy, uninstall, and customize the governance policy framework on my
+   managed clusters.
+
+### Architecture
+
+The governance policy addon controller manages installations of policy addons on managed clusters by
+using
+[ManifestWorks](https://github.com/open-cluster-management-io/api/blob/main/docs/manifestwork.md).
+The addons can be enabled, disabled, and configured via their `ManagedClusterAddOn` resources.
+
+(For more information on the addon framework, see the
+[addon-framework enhancement/design](https://github.com/open-cluster-management-io/enhancements/tree/main/enhancements/sig-architecture/8-addon-framework).)
+
+The addons managed by this controller are:
+
+- The "config-policy-controller" consisting of the
+  [Configuration Policy Controller](https://github.com/open-cluster-management-io/config-policy-controller).
+- The "governance-policy-framework" consisting of the
+  [Policy Spec Sync](https://github.com/open-cluster-management-io/governance-policy-spec-sync), the
+  [Policy Status Sync](https://github.com/open-cluster-management-io/governance-policy-status-sync),
+  and the
+  [Policy Template Sync](https://github.com/open-cluster-management-io/governance-policy-template-sync).
+
+For example, this manifest deploys the Configuration Policy Controller to a managed cluster called
+`my-managed-cluster` to the `open-cluster-management-agent-addon` namespace:
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedClusterAddOn
+metadata:
+  name: config-policy-controller
+  namespace: my-managed-cluster
+spec:
+  installNamespace: open-cluster-management-agent-addon
+```
+
+To modify the image used by the Configuration Policy Controller on this managed cluster, you can add
+an annotation like:
+
+```shell
+kubectl -n my-managed-cluster annotate ManagedClusterAddOn config-policy-controller addon.open-cluster-management.io/values='
```

Comment:
i suppose it's generally not recommended to extend structurized annotations on kubernetes resources. can we flatten them to e.g. `policy.addon.open-cluster-management.io/config-policy-controller-image: <...>`, etc.?

Reply:
Right, I see the `values` annotation as a way to get at *any* value in the helm chart's values.yaml. But it's not an ideal user experience. I'd like to move some of the useful things into separate annotations, for example an annotation for log levels.

Reply:
FYI @zhiweiyin318

Reply:
yes, you can expend any ways to override the values for your templates. [WithGetValuesFuncs](https://github.com/open-cluster-management-io/addon-framework/blob/edd160091e30b9f3c19a7a7b6ac31d0c770c2a91/pkg/addonfactory/addonfactory.go#L61)  can input a slice of GetValues Funcs.
the annotation `addon.open-cluster-management.io/values=` is a default GetValues Func for ACM klusterlet-addons, because import-controller needs to configure it with override images, node Selector and Proxy env.